src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs

```diff
@@ -10,9 +10,41 @@ namespace Microsoft.CodeAnalysis.CSharp
     internal sealed partial class LocalRewriter
     {
         private BoundStatement RewriteNullChecking(BoundBlock block)
+        {
+            var statementList = ConstructNullCheckedStatementList(_factory.CurrentFunction.Parameters, block.Statements);
+            if (statementList.IsDefaultOrEmpty)
+            {
+                return block;
+            }
+            return block.Update(statementList);
+        }
+
+        private BoundLambda RewriteNullChecking(BoundLambda lambda)
+        {
+            var statementList = ConstructNullCheckedStatementList(_factory.CurrentFunction.Parameters, lambda.Body.Statements);
+            if (statementList.IsDefaultOrEmpty)
+            {
+                return lambda;
+            }
+            var newBody = _factory.Block(lambda.Body.Locals, statementList);
+            return lambda.Update(lambda.UnboundLambda, lambda.Symbol, newBody, lambda.Diagnostics, lambda.Binder, lambda.Type);
+        }
+
+        private BoundLocalFunctionStatement RewriteNullChecking(BoundLocalFunctionStatement localFunctionStatement)
+        {
+            var statementList = ConstructNullCheckedStatementList(_factory.CurrentFunction.Parameters, localFunctionStatement.Body.Statements);
+            if (statementList.IsDefaultOrEmpty)
+            {
+                return localFunctionStatement;
+            }
+            var newBody = _factory.Block(localFunctionStatement.BlockBody.Locals, statementList);
+            return localFunctionStatement.Update(localFunctionStatement.Symbol, newBody, localFunctionStatement.ExpressionBody);
```

Comment:
>localFunctionStatement.ExpressionBody [](start = 89, length = 37)

Is this the original `ExpressionBody`? Are we testing expression body local functions?

Reply:
I believe it is the original. How can I test the expression body ones directly?

Reply:
With a local function such as:
```
object Local(object arg!) => arg;
```

---
In reply to: [298351427](https://github.com/dotnet/roslyn/pull/36832#discussion_r298351427) [](ancestors = 298351427)

Reply:
This local function doesn't seem to even make it into VisitLocalFunctionStatement, but it lowered appropriately..